cluster/gce/util.sh

```diff
@@ -2359,7 +2359,8 @@ function create-nodes() {
     gcloud compute instance-groups managed wait-until-stable \
         "${group_name}" \
         --zone "${ZONE}" \
-        --project "${PROJECT}" || true;
+        --project "${PROJECT}" \
+        --timeout "${MIG_WAIT_UNTIL_STABLE_TIMEOUT}" || true;
   done
```

Comment:
While I'm perfectly fine with not waiting forever for mig to become stable, I don't think it's a good idea to continue test with only partial set of nodes: either test will fail later because it won't be able to do regular job (like scheduling 30*5k pods on nodes it has) or we will have false feeling that tests are passing (while what we call '5k node test' can be passing with only ~100 nodes).

Reply:
Good point. Let me add few thoughts:

- I think if we succeeded to start let's say 4k out of 5k, it's still better to run tests and get some scalability signal rather than wasting the run altogether
- I'm not sure if the test would (should) fail in such a scenario as iirc we calculate #pods based on #nodes registered. If there are some other places where there's dependency of the test on NUM_NODES, we probably should revisit them too
- If we're worried about not catching such issues (i.e only 4k started out of 5k), we can probably just report it as an error/flake and proceed

Reply:
How often that problem happens?

I think last of your points should be addressed before doing that...

Reply:
> How often that problem happens?

I remember this happening a few times in the past (when we could only start about 3k-4k nodes due to zone being busy).

> I think last of your points should be addressed before doing that...

It seems like we're currently already catching error if not enough nodes are ready during [validate-cluster](https://github.com/kubernetes/kubernetes/blob/c6a852fe74890c9d890e8c35e7e4dcf2d5b84145/cluster/validate-cluster.sh#L138-L139). So basically this change would currently assure that we don't wait arbitrarily long for MIGs, but go ahead and fail fast. For my 1st point, we can change our large-scale jobs to have higher tolerance if needed (but still report a flake through test-infra).

Reply:
Ok - the validate-cluster bit is good to know.

Re increasing tollerance - I don't think we should allow for higher toleration. If we're running 5k-node test, then we don't want to run with 3k or 4k nodes, because that will make results incomparable, etc.

But not waiting long enough probably make sense given that we will fail right after that.